src/libdredd/src/mutation_replace_unary_operator.cc

```diff
@@ -218,17 +285,22 @@ std::string MutationReplaceUnaryOperator::GenerateMutatorFunction(
     mutant_offset++;
   }
 
-  new_function << "  return ";
-  if (IsPrefix(unary_operator_.getOpcode())) {
-    new_function
-        << clang::UnaryOperator::getOpcodeStr(unary_operator_.getOpcode()).str()
-        << arg_evaluated + ";\n";
-  } else {
-    new_function
-        << arg_evaluated
-        << clang::UnaryOperator::getOpcodeStr(unary_operator_.getOpcode()).str()
-        << ";\n";
+  const clang::BuiltinType* exprType =
+      unary_operator_.getType()->getAs<clang::BuiltinType>();
+  if ((exprType->isBooleanType() &&
+       unary_operator_.getOpcode() != clang::UnaryOperatorKind::UO_LNot) ||
+      (exprType->isInteger() && !exprType->isBooleanType() &&
+       !unary_operator_.isLValue())) {
+    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                 << mutant_offset << ")) return !(" << getExpr(ast_context)
+                 << ");\n";
+    mutant_offset++;
```

Comment:
From the diff it wasn't obvious to me why this code had changed (but I'm getting a bit tired!)

Reply:
I think the diff has messed up a little here. The changed code is for inserting the `!` unary operator where as the code that was removed was the previous way for returning the unmutated expression. The latter has now been changed to use GetExpr and is further down.